File: content/snippets/cpp/language_basics_command_line_args.cpp
#include <iostream>
#include <string>

int main(int argc, char** argv) {
    std::string name = argc > 1 ? argv[1] : "world";
    bool excited = argc > 2 && std::string(argv[2]) == "--excited";

    std::cout << "Hello, " << name;
    if (excited) {
        std::cout << "!";
    }
    std::cout << "\n";
    return 0;
}

File: content/snippets/python/language_basics_command_line_args.py
#!/usr/bin/env python3
import sys


def main() -> None:
    name = sys.argv[1] if len(sys.argv) > 1 else "world"
    excited = len(sys.argv) > 2 and sys.argv[2] == "--excited"

    message = f"Hello, {name}"
    if excited:
        message += "!"
    print(message)


if __name__ == "__main__":
    main()
